src/conn/conn_oligarch.c

```diff
@@ -8,192 +8,97 @@
 
 #include "wt_internal.h"
 
-static WT_THREAD_RET
-__oligarch_metadata_watcher(void *arg)
+/*
+ * __disagg_pick_up_checkpoint --
+ *     Pick up a new checkpoint.
+ */
+static int
+__disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t checkpoint_id)
 {
     WT_CONNECTION_IMPL *conn;
     WT_CURSOR *md_cursor;
     WT_DECL_ITEM(item);
     WT_DECL_RET;
-    WT_FH *md_fh;
-    WT_SESSION_IMPL *session;
-    char buf[4096], *cfg_ret, *md_path,
-      *new_md_value; /* TODO the 4096 puts an upper bound on metadata entry length */
-    const char *value, *cfg[3];
-    int i;
+    WT_SESSION_IMPL *internal_session;
+    char buf[4096], *cfg_ret,
+      *metadata_value_cfg; /* TODO the 4096 puts an upper bound on metadata entry length */
+    const char *cfg[3], *current_value, *metadata_key, *metadata_value;
     size_t len;
-    wt_off_t last_sep, last_sz, name_ptr, new_sz;
+    wt_off_t sep;
 
-    md_fh = NULL;
-    session = (WT_SESSION_IMPL *)arg;
     conn = S2C(session);
-    md_fh = NULL;
-    memset(buf, 0, 4096);
 
-    WT_ERR(__wt_scr_alloc(session, 4096, &item));
-
-    len = strlen(conn->iface.stable_prefix) + strlen(WT_OLIGARCH_METADATA_FILE) + 2;
-    WT_ERR(__wt_calloc_def(session, len, &md_path));
-    WT_ERR(
-      __wt_snprintf(md_path, len, "%s/%s", conn->iface.stable_prefix, WT_OLIGARCH_METADATA_FILE));
+    internal_session = NULL;
+    md_cursor = NULL;
+    metadata_key = NULL;
+    metadata_value = NULL;
+    metadata_value_cfg = NULL;
 
-    /*
-     * TODO this loop is currently needed so that we don't ENOENT out of the watcher thread while
-     * the primary is starting up and hasn't yet created the shared file.
-     */
-    for (i = 0; i < 1000; i++) {
-        ret = __wt_open(session, md_path, WT_FS_OPEN_FILE_TYPE_DATA, WT_FS_OPEN_FIXED, &md_fh);
-        if (ret == ENOENT)
-            __wt_sleep(1, 0);
-        else if (ret == 0)
-            break;
-        else
-            WT_ERR(ret);
-    }
-    if (i == 1000)
-        WT_ERR(WT_NOTFOUND);
-    WT_ERR(__wt_filesize(session, md_fh, &last_sz));
+    WT_ERR(__wt_scr_alloc(session, 4096, &item));
 
-    /* TODO this will need to handle multiple tables */
-    for (;;) {
-        __wt_sleep(0, 1000);
-        if (F_ISSET(conn, WT_CONN_CLOSING))
+    /* Only followers are allowed to pick up new checkpoints. */
+    if (conn->oligarch_manager.leader)
+        WT_ERR(EINVAL);
+
+    /* Read the checkpoint metadata from the special metadata page. */
+    WT_ERR(__wt_disagg_get_meta(session, 0, checkpoint_id, item));
+
+    if (item->size >= sizeof(buf))
+        WT_ERR(EINVAL);
+    memcpy(buf, item->data, item->size);
+    buf[item->size] = '\0';
+    if (item->size > 0 && buf[item->size - 1] == '\n')
+        buf[item->size - 1] = '\0';
+
+    /* Parse out the key and the new checkpoint config value. */
+    metadata_key = buf;
+    for (sep = (wt_off_t)item->size; sep >= 0; sep--)
```

Comment:
I'm wondering why this goes from the back to the front to search for '|'?  I would think that having a '|' would be much rarer than having a '|' in the value, so maybe front to back is better?

Also maybe '\n' is a better separator for this case? (could be done later)